src/_pytest/pathlib.py

```diff
@@ -539,6 +539,10 @@ def import_path(
         except CouldNotResolvePathError:
             pass
         else:
+            # If the given module name is already in sys.modules, do not import it again.
+            with contextlib.suppress(KeyError):
```

Comment:
just taking a look again, i believe this should trigger a import mismatch if the file of the loaded module and the current path are different